tests/unit/package_managers/test_pip.py

```diff
@@ -2864,58 +2870,59 @@ def test_process_package_links_noncanonical_version(
         else:
             actual_version = noncanonical_version
 
-        links = self.mock_html_links(
-            f'<a href="../foo-{actual_version}.tar.gz">foo-{actual_version}.tar.gz</a>',
+        mock_requirement = self.mock_requirement(
+            "foo", "pypi", version_specs=[("==", requested_version)]
         )
-
-        assert pip._process_package_links(links, "foo", requested_version) == [
-            {
-                "name": "foo",
-                "version": actual_version,
-                "url": f"../foo-{actual_version}.tar.gz",
-                "filename": f"foo-{actual_version}.tar.gz",
-                "yanked": False,
-            }
-        ]
-
-    def test_process_package_links_not_sdist(self) -> None:
-        """Test that links for files that are not sdists are ignored."""
-        links = self.mock_html_links(
-            '<a href="../foo-1.0.whl">foo-1.0.whl</a>',
-            '<a href="../foo-1.0.egg">foo-1.0.egg</a>',
+        mock_get_project_page.return_value = pypi_simple.ProjectPage(
+            "foo",
+            [self.mock_pypi_simple_package(filename="foo.tar.gz", version=actual_version)],
+            None,
+            None,
         )
-        assert pip._process_package_links(links, "foo", "1.0") == []
 
-    @pytest.mark.parametrize("requested_version", ["2.0", "1.0.a1", "1.0.post1", "1.0.dev1"])
-    def test_process_package_links_wrong_version(self, requested_version: str) -> None:
-        """Test that links for files with different version are ignored."""
-        links = self.mock_html_links(
-            '<a href="../foo-1.0.tar.gz">foo-1.0.tar.gz</a>',
-        )
-        assert pip._process_package_links(links, "foo", requested_version) == []
+        source, _ = pip._process_package_distributions(mock_requirement, rooted_tmp_path)
+        assert source.version == requested_version
```

Comment:
IMO we should test wheels as well here, the version canonicalization should work the same way for sdists and wheels